polygeist/lib/Conversion/PolygeistToLLVM/PolygeistToLLVM.cpp

```diff
@@ -103,8 +459,10 @@ struct BaseSubIndexOpLowering : public ConvertOpToLLVMPattern<SubIndexOp> {
               assert(t.getBody().size() == 1 && "Expecting single member type");
               currType = t.getBody()[0];
             })
-            .Case<LLVM::LLVMArrayType, LLVM::LLVMPointerType>(
+            .Case<LLVM::LLVMArrayType>(
                 [&](auto t) { currType = t.getElementType(); })
+            .Case<LLVM::LLVMPointerType>(
+                [&](auto t) { assert(false && "Pointer type not allowed"); })
```

Comment:
Why not allowed to have struct of pointer?

Reply:
It is allowed to have a struct of pointer (`struct<ptr>`) and this change doesn't inhibit that. 

What doesn't work anymore due to opaque pointers, is a struct behind a pointer nested inside a struct, i.e. `struct<ptr<struct>>`. 

Before opaque pointers, it was possible to access the `i64` element in this example: `struct<ptr<struct<i64, double>>`, because we could inspect the type of the struct pointed to. 

With opaque pointer, all we will see is `struct<ptr>`, as the element type of the pointer (`struct<i64, double>` in the example) isn't present with opaque pointers, so we need to stop inspection here. Hence the change here.